new-packages/ts-project/src/state.ts

```diff
@@ -348,6 +361,43 @@ function extractEdgeGroup(
               registerActionBlocks(ctx, blocks, edge.data.actions);
               return;
             }
+            case 'internal': {
+              if (seenInternalProp) {
+                // `reenter` was already seen
+                ctx.errors.push({
+                  type: 'property_mixed',
+                });
+                return;
+              }
+              seenInternalProp = true;
+
+              if (findProperty(ctx, ts, element, 'reenter')) {
+                return;
+              }
+
+              const value = getJsonValue(ctx, ts, prop.initializer);
```

Comment:
nit: I'd rather use `getJsonValue` for when we want to extract a JSON value like we do with metadata, context, etc. We could directly check for AST kind here which is more specific IMO.

Reply:
good point, gonna do this change